arkouda/segarray.py

```diff
@@ -250,7 +250,9 @@ def __getitem__(self, i):
         if isSupportedInt(i):
             start = self.segments[i]
             end = self.segments[i] + self.lengths[i]
-            return self.values[start:end].to_ndarray()
```

Comment:
With the iterative approach used in this PR, I can see how the `.to_ndarray()` would break things. However, as I explain in another comment, I think we should move away from the iterative approach, and I also think there are good reasons for keeping the `.to_ndarray()` here:
1. For other arkouda objects, a single-element access returns a python/numpy object (e.g. a scalar integer) instead of an arkouda array, and it would be nice for SegArrays to keep consistent with that design.
2. Related: I think when a DataFrame tries to display itself, it gets single elements one at a time from each column, and it expects that those elements will be python/numpy objects, not arkouda objects.

Reply:
Understood. I will be sure to change this here once #1218 has been worked and this is ready for review again.

Reply:
does this still need to be changed back?